src/base/searchengine.h

```diff
@@ -38,10 +40,119 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
```

Comment:
In `base/`.
But I have another look at it. Why bother to use this template? Why not just make one generic Version class?
We can have fixed format version: major.minor.release.patch. And we can represent it as follows: major[.minor[.release[.patch]]], where any omitted rest is equal to zero.